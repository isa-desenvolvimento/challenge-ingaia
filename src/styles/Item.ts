import styled from 'styled-components'
import Image from 'next/image'

export const Container = styled.div`
  height: 223px;
  width: 223px;
  position: relative;
  border-radius: 8px;

  &:hover {
    border: 2px solid rgba(251, 255, 5, 0.794);
  }

  img {
    border-radius: 8px;
  }
`
export const ImageContainer = styled(Image)`
  max-width: 100%;
  max-height: 100%;
  border-radius: inherit;
  filter: ${({ title }) => (title === 'Dead' ? 'grayscale(1)' : 'none')};
`
export const Description = styled.div`
  backdrop-filter: blur(0.1rem);
  background: rgba(49, 49, 49, 0.9);
  position: absolute;
  width: 100%;
  bottom: 0;
  color: white;
  padding: 1rem;
  border-bottom-right-radius: inherit;
  border-bottom-left-radius: inherit;
`
