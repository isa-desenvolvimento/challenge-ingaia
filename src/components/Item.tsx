import * as S from '../styles/Item'
import { useTranslationContext } from '../locales/translationContext'

type ItemsProps = {
  name: string
  image: string
  species: string
  status: string
  width?: string
  height?: string
  onclick?: () => void
}

export default function ItemList({
  name,
  species,
  image,
  status,
  width = '283',
  height = '238',
  onclick
}: ItemsProps) {
  const translation = useTranslationContext()

  return (
    <S.Container onClick={onclick}>
      <S.ImageContainer
        src={image}
        alt={`${translation.IMAGE_EPISODE} ${name}`}
        width={width}
        height={height}
        objectFit="cover"
        title={status}
        loading="lazy"
        role="image"
      />
      <S.Description>
        <div>
          <h2>{name}</h2>
          <span> {species} </span>
        </div>
      </S.Description>
    </S.Container>
  )
}
