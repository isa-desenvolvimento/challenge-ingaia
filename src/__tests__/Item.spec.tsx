import { render } from '@testing-library/react';
import Item from '../components/Item';

jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {
    images: {
      domains: ['cdn.example.com'],
    },
  },
}));

jest.mock('next/image', () => ({ src, alt }) => <img src={src} alt={alt} />);

describe('<Item />', () => {
  const props = {
    thumbnail: 'http://http2.mlstatic.com/D_844171-MLA41757309022_052020-O.jpg',
    address: {
      city_name: 'city_name_test',
    },
    price: 100,
    id: 'test',
    title: 'test',
    shipping: {
      free_shipping: false,
    },
  };

  it('should render component Item children', () => {
    props.shipping.free_shipping = true;

    const { getByTestId } = render(
      <Item {...props} />,
    );

    expect(getByTestId('txt_price_item')).toHaveTextContent('R$ 100,00');
    expect(getByTestId('txt_title_item')).toHaveTextContent('test');
    expect(getByTestId('txt_address_item')).toHaveTextContent('city_name_test');
    expect(getByTestId('ic_shipping_item')).toBeVisible();
  });

  it('should render snapshot', () => {
    const { container } = render(
      <Item {...props} />,
    );
    expect(container).toMatchSnapshot();
  });
});
