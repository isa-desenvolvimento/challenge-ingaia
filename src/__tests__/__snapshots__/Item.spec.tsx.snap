// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render snapshot 1`] = `
<div>
  <div
    class="container"
    id="test"
  >
    <div
      class="imageContainer"
    >
      <img
        alt="test"
        src="http://http2.mlstatic.com/D_844171-MLA41757309022_052020-O.jpg"
      />
    </div>
    <div
      class="containerDetails"
    >
      <div
        class="priceContainer"
      >
        <h2
          data-testid="txt_price_item"
        >
          R$Â 100,00
        </h2>
        <div
          class="shipping"
          data-testid="ic_shipping_item"
        />
      </div>
      <span
        data-testid="txt_title_item"
      >
        test
      </span>
    </div>
    <span
      class="address"
      data-testid="txt_address_item"
    >
      city_name_test
    </span>
  </div>
</div>
`;
